packages/error-reporting/system-test/testAuthClient.js

```diff
@@ -357,3 +359,69 @@ describe('Expected Behavior', function() {
     });
   });
 });
+
+describe('error-reporting', function() {
+  const TIMESTAMP = Date.now();
+  const BASE_NAME = 'error-reporting-system-test';
+  function buildName(suffix) {
+    return [TIMESTAMP, BASE_NAME, suffix].join('_');
+  }
+
+  const SERVICE_NAME = buildName('service-name');
+  const SERVICE_VERSION = buildName('service-version');
+
+  var errors;
+  var transport;
+  before(function() {
+    errors = require('../src/index.js')({
+      ignoreEnvironmentCheck: true,
+      serviceContext: {
+        service: SERVICE_NAME,
+        version: SERVICE_VERSION
+      }
+    });
+    transport = new ErrorsApiTransport(errors._config, errors._logger);
+  });
+
+  after(function(done) {
+    transport.deleteAllEvents(function(err) {
+      assert.ifError(err);
+      done();
+    });
+  });
+
+  it('Should correctly publish errors', function(done) {
+    this.timeout(TIMEOUT * 2);
+    var errorId = buildName('message');
+    errors.report(new Error(errorId), function(err, response, body) {
+      assert.ifError(err);
+      assert(isObject(response));
+      assert.deepEqual(body, {});
+
+      setTimeout(function() {
+        transport.getAllGroups(function(err, groups) {
+          assert.ifError(err);
+          assert.ok(groups);
+
+          var matchedErrors = groups.filter(function(errItem) {
+            return errItem && errItem.representative &&
+              errItem.representative.message.startsWith('Error: ' + errorId);
+          });
+
+          // The error should have been reported exactly once
+          assert.strictEqual(matchedErrors.length, 1);
+          var errItem = matchedErrors[0];
+          assert.ok(errItem);
+          assert.equal(errItem.count, 1);
+          var rep = errItem.representative;
+          assert.ok(rep);
+          var context = rep.serviceContext;
+          assert.ok(context);
```

Comment:
If time stamps are recorded when errors are published, those could be used to ensure we are observing the correct error.